lib/Controller/RoomController.php

```diff
@@ -1108,6 +1109,17 @@ public function addParticipantToRoom(string $newParticipant, string $source = 'u
 		$participantsToAdd = [];
 
 		if ($source === 'users') {
+
+			//Check if the newParticipant is banned
+			try {
+				$this->banService->throwIfActorIsBanned($this->room, $newParticipant);
```

Comment:
I'd prefer adding a method to BanService that passes on the result of `findForBannedActorAndRoom` and calling it directly with the correct actor_type and actor_id

Also this should not log